### PR TITLE
Create better hearings data for demo

### PR DESCRIPTION
**Pull request by rubaiyat22**
Relates to https://vajira.max.gov/browse/CASEFLOW-990

### Description
Based on discussion with Product and Design on this [thread](https://dsva.slack.com/archives/C01155MJGF2/p1614632661063300?thread_ts=1614272047.034900&cid=C01155MJGF2), this PR refactors existing hearings seed script to create better data, specifically focusing on varied hearing times, varied judges for hearing days, varied poas, and varied Veteran/Claimant data. 
This also creates varied request issues for unscheduled hearings.
Additionally this PR adds user `BVASYELLOW` to hearings admin.
#### Testing Plan
1. run `make reset`
2. login as `BVASYELLOW` and go to http://localhost:3000/hearings/schedule
3. Poke around to see that the data is varied

### CODE REVIEW

File: app/models/serializers/work_queue/legacy_appeal_serializer.rb [hunk 1]
```diff
@@ -65,9 +65,9 @@ class WorkQueue::LegacyAppealSerializer
 
   attribute :regional_office do |object|
     {
-      key: object.regional_office.key,
-      city: object.regional_office.city,
-      state: object.regional_office.state
+      key: object.regional_office&.key,
```
Comment (rubaiyat22) [inline app/models/serializers/work_queue/legacy_appeal_serializer.rb:68]: unrelated change

File: db/seeds/tasks.rb [hunk 2]
```diff
@@ -388,6 +388,7 @@ def create_different_hearings_tasks
         30.times do
           appeal = create(
             :appeal,
+            :with_request_issues,
```
Comment (rubaiyat22) [inline db/seeds/tasks.rb:391]: this data is populated in the https://hearings.caseflowdemo.com/hearings/schedule/assign?tab=amaAssignHearingsTab&regional_office_key=RO43&page=1 table. I just made this small change for there to be varied request issues. No other changes were made for the table data.

**Review by charleyf — COMMENTED** [review]
These changes definitely improve the data that we seed for hearings and as far as I can tell they are good. My unfamiliarity with how the data should work has made reviewing this slow going and I'm not sure I understood enough to 'approve' this PR.

I would appreciate if @tomas-nava could also review these changes before they are merged?

Ordered by 'likely interest to the reader' here's what I did:
- Confirmed that we do end up with HearingDay(s) with AMA hearings, legacy hearings, and a mix:
`HearingDay.all.map { |d| d.open_hearings }.flatten.pluck(:vacols_id, :uuid)`
- Made a readability change to the loop that creates hearing days (pushed)
- Did a lot of research:
  - Kat dealing with factories: https://gist.github.com/lomky/0653b27d528ac975ac3ce4757bb1bca8
  - PR improving factories around MTV(motion to vacate) https://github.com/department-of-veterans-affairs/caseflow/pull/15616#
  - db/seeds/mtv.rb
  - Yoom finding out that creating a claimant makes an appeal: https://dsva.slack.com/archives/CJL810329/p1598466059147100?thread_ts=1598462259.130100&cid=CJL810329
  - Tomas saying we should try to avoid create in factories: https://dsva.slack.com/archives/C2ZAMLK88/p1564008758048500?thread_ts=1563914481.037900&cid=C2ZAMLK88
  - A thread about factory users: https://dsva.slack.com/archives/C2ZAMLK88/p1581107452235300

File: db/seeds/hearings.rb [hunk 3]
```diff
@@ -199,25 +242,37 @@ def create_veteran(veteran_file_number)
...
-        bfcorkey: "5678#{id}",
-        correspondent: create(:correspondent, stafkey: "5678#{id}")
+        bfkey: @bfkey.to_s,
+        bfcorkey: @bfcorkey.to_s,
+        correspondent: create(:correspondent, stafkey: @bfcorkey.to_s)
       )
     end
 
     def create_former_travel_currently_virtual_requested_legacy_appeals
       ro_key = "R" # virtual
```
Comment (tomas-nava) [inline db/seeds/hearings.rb:267]: I know this isn't new to this PR, but `"R"` is not a valid value for `closest_regional_office` on an appeal. It's probably harmless, but maybe on line 283 we should set `closest_regional_office` to `nil` if `ro_key` is `"R"`?

**Review by tomas-nava — APPROVED** [review]
approve!

left a few non-blocking suggestions

File: db/seeds/hearings.rb (hunk 3 above)
Comment (rubaiyat22) [reply db/seeds/hearings.rb:267]: That was a oversight on my part. I set `closest_regional_office` to be `nil`
